Demo/app/src/main/java/com/krunal/demo/kotlinPractice/oop/Inheritance.kt

```diff
@@ -0,0 +1,66 @@
+package com.krunal.demo.kotlinPractice.oop
+
+import kotlin.math.max
+
+fun main() {
+    var animal: Animal = object : Animal(name = "Tom") {
+        override fun canSecure(): Boolean {
+            return false
+        }
+    }
+    println(animal.speak())
+    animal = Dog("dog")
+    println(animal.speak())
+    animal = Tommy()
+    println(animal::class.java)
+    println(animal.sleepTime)
+
+    var derived: Base// = Derived("Hello", "World")
```

Comment:
remove commented code if not needed

Reply:
Some commented codes are for reference and indicate other possibilities.
For e.g. here
```kotlin
    var derived: Base// = Derived("Hello", "World")
    derived = Derived("Hello", "World")
    println(derived.name)
    println(derived.lastName)
```
if we directly assign `Derived class` object to `derived` then members of `Derived` won't be accessible. On the other hand if we declare and assign separately members of derived will be accessible.